### ISSUE TITLE
Uninitialized(?) error string (sometimes)

### ISSUE THREAD

**Issue by robkar** [chrchang/plink-ng#132]
Hi, this is an odd one. I was running --glm in plink2, and noticed that for some variants on chrX, the "err/ERRCODE" column for the automatic SEX covariate line held an arbitrary piece of text from the manual/--help. 
The corresponding marker lines where I noticed this had the ERRCODE "CONST_ALLELE,REF" (and from a quick look this seems to happen for most chrX markers with that ERRCODE).

When adding --chr X to the command and keeping all else constant, the same covariate lines look OK (empty ERRCODE field for the covariate column).
Extracting just one autosomal and one chrX marker previously showing the extra output had the same effect - no manual text in the output.
The behavior was not fixed by reducing threads from many to 1, but the output changed, so runs with the same number of threads produced the same number of lines of output (different for different number of threads).

I could not easily create a reproducible example without sharing non-sharable data, so hoping this description can point you in the right direction at least. Let me know if there is anything else I can do to help find the bug location.

## Full run:

```
PLINK v2.00a2LM 64-bit Intel (12 Nov 2019)
Options in effect:
  --bfile Data/DerivedData/QC/04_preqc/QE-1466_pqc1
  --glm firth-fallback cols=+a1countcc,+totallelecc,+err
  --memory 8000
  --out full
  --pheno zyg_unrel.pheno
  --threads 1

Hostname: vector.meb.ki.se
Working directory: /nfs/projects/STR_GWAS/gsa
Start time: Fri Nov 15 15:22:47 2019

Random number seed: 1573827767
1483177 MiB RAM detected; reserving 8000 MiB for main workspace.
Using 1 compute thread.
15294 samples (8778 females, 6516 males; 15294 founders) loaded from
Data/DerivedData/QC/04_preqc/QE-1466_pqc1.fam.
700078 variants loaded from Data/DerivedData/QC/04_preqc/QE-1466_pqc1.bim.
1 binary phenotype loaded (7417 cases, 4621 controls).
Calculating allele frequencies... done.
--glm logistic-Firth hybrid regression on phenotype 'ZYG': done.
Results written to full.ZYG.glm.logistic.hybrid .

End time: Fri Nov 15 15:34:23 2019
```

## Full run results (extract):

```
#CHROM	POS	ID	REF	ALT	A1	A1_CASE_CT	A1_CTRL_CT	CASE_ALLELE_CT	CTRL_ALLELE_CT	FIRTH?	TEST	OBS_CT	OR	LOG(OR)_SE	Z_STAT	P	ERRCODE
X	153774273	seq-rs398123552	A	G	G	0	0	14830	9242	N	ADD	12036	NA	NA	NA	NA	CONST_ALLELE,REF
X	153774273	seq-rs398123552	A	G	G	0	0	14830	9242	N	SEX	12036	NA	NA	NA	NA	esent each dosage.  'format=1' normally indicates a single 0..2 A1
      expected count; 'dose1' modifies this to a 0..1 frequency.  'format=2'
      indicates a 0..1 homozygous A1 likelihood followed by a 0..1 het
      likelihood.  'format=3' indicates 0..1 hom A1, 0..1 het, 0..1 hom A2.
      'format=infer' (the default) infers the format from the number of columns
      in the first nonheader line.


X	153780224	chrX-153780224	A	.	.	0	0	14834	9240	N	ADD	12037	NA	NA	NA	NA	CONST_ALLELE,REF
X	153780224	chrX-153780224	A	.	.	0	0	14834	9240	N	SEX	12037	NA	NA	NA	NA	mat=1' normally indicates a single 0..2 A1
      expected count; 'dose1' modifies this to a 0..1 frequency.  'format=2'
      indicates a 0..1 homozygous A1 likelihood followed by a 0..1 het
      likelihood.  'format=3' indicates 0..1 hom A1, 0..1 het, 0..1 hom A2.
      'format=infer' (the default) infers the format from the number of columns
      in the first nonheader line.


X	153820769	X:153820769-T-C	T	C	C	224	137	14824	9238	N	ADD	12031	1.01193	0.0908699	0.130531	0.896146	.
X	153820769	X:153820769-T-C	T	C	C	224	137	14824	9238	N	SEX	12031	0.933907	0.0380153	-1.7987	0.0720659	.
X	153826990	rs112477758	G	A	A	0	0	14804	9230	N	ADD	12017	NA	NA	NA	NA	CONST_ALLELE,REF
X	153826990	rs112477758	G	A	A	0	0	14804	9230	N	SEX	12017	NA	NA	NA	NA	(the default) infers the format from the number of columns
      in the first nonheader line.


X	153827637	rs4898389	A	G	G	1929	1183	14834	9240	N	ADD	12037	1.01227	0.0332202	0.367053	0.71358	.
X	153827637	rs4898389	A	G	G	1929	1183	14834	9240	N	SEX	12037	0.933488	0.0380064	-1.81093	0.0701513	.
```

## Run with fewer markers:

```
PLINK v2.00a2LM 64-bit Intel (12 Nov 2019)
Options in effect:
  --bfile Data/DerivedData/QC/04_preqc/QE-1466_pqc1
  --d @
  --glm firth-fallback cols=+a1countcc,+totallelecc,+err
  --memory 8000
  --out twosnps
  --pheno zyg_unrel.pheno
  --snps seq-rs398123552,GSA-1:726912
  --threads 1

Hostname: vector.meb.ki.se
Working directory: /nfs/projects/STR_GWAS/gsa
Start time: Fri Nov 15 14:50:15 2019

Random number seed: 1573825815
1483177 MiB RAM detected; reserving 8000 MiB for main workspace.
Using 1 compute thread.
15294 samples (8778 females, 6516 males; 15294 founders) loaded from
Data/DerivedData/QC/04_preqc/QE-1466_pqc1.fam.
700078 variants loaded from Data/DerivedData/QC/04_preqc/QE-1466_pqc1.bim.
1 binary phenotype loaded (7417 cases, 4621 controls).
--snps: 2 variants remaining.
Calculating allele frequencies... done.
2 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'ZYG': done.
Results written to twosnps.ZYG.glm.logistic.hybrid .

End time: Fri Nov 15 14:50:17 2019
```

## Results:

```
#CHROM	POS	ID	REF	ALT	A1	A1_CASE_CT	A1_CTRL_CT	CASE_ALLELE_CT	CTRL_ALLELE_CT	FIRTH?	TEST	OBS_CT	OR	LOG(OR)_SE	Z_STAT	P	ERRCODE
1	726912	GSA-1:726912	A	.	.	0	0	14594	9118	N	ADD	11856	NA	NA	NA	NA	CONST_ALLELE,REF
X	153774273	seq-rs398123552	A	G	G	0	0	14830	9242	N	ADD	12036	NA	NA	NA	NA	CONST_ALLELE,REF
X	153774273	seq-rs398123552	A	G	G	0	0	14830	9242	N	SEX	12036	NA	NA	NA	NA	
```

**Comment by chrchang** [comment]
Thanks for reporting this.  I've replicated this problem; will try to post a fix tonight.

**Comment by chrchang** [comment]
The bug should be fixed by 1bcfd40 .

**Comment by robkar** [comment]
Yes, Nov 15 binaries work for me. Many thanks!
